hooks/src/index.js

```diff
@@ -366,6 +366,8 @@ function invokeEffect(hook) {
 	// and move the currentComponent away.
 	const comp = currentComponent;
 	hook._cleanup = hook._value();
+	hook._args = hook._pendingArgs;
```

Comment:
This measure is taken so that when a hook is scheduled but cancelled due to Suspense or unmount we aren't eagerly changing the _args, we are now introducing an _pending state that will only be applied when the cb executes.

Reply:
Oh nice! Good call. Does this have any observable change we could test?

Reply:
Thinking more about this change, does it mean we would only run effects if they changed from the original deps? So if I initiate two rerenders quickly that both suspend and the final rerender matches the original deps, then we wouldn't rerun effects here?

I'm thinking something like this:
```
Initial render deps: [1]
1st rerender deps: [2] // Suspends
2nd rerender deps: [1] // Happens before 1st rerender unsuspends. With this change would effects run with these args?
// Should they? Maybe they shouldn't?
```
```